### PR TITLE
Implement booleanWithin

### PR DESCRIPTION
**Pull request by jsiedentop**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -237,7 +237,7 @@ Any new benchmarks must be named `*_benchmark.dart` and reside in the
 - [x] [booleanParallel](https://github.com/dartclub/turf_dart/blob/main/lib/src/booleans/boolean_parallel.dart)
 - [x] [booleanPointInPolygon](https://github.com/dartclub/turf_dart/blob/main/lib/src/booleans/boolean_point_in_polygon.dart)
 - [x] [booleanPointOnLine](https://github.com/dartclub/turf_dart/blob/main/lib/src/booleans/boolean_point_on_line.dart)
-- [ ] booleanWithin
+- [x] booleanWithin
```
Comment (lukas-h) [inline README.md:240]: we need the url to the file here

Comment (jsiedentop) [reply README.md]: Done. I don't know how I could have missed that.

File: lib/src/booleans/boolean_contains.dart [hunk 2]
```diff
@@ -27,135 +28,48 @@ bool booleanContains(GeoJSONObject feature1, GeoJSONObject feature2) {
 
   var coords1 = (geom1 as GeometryType).coordinates;
   var coords2 = (geom2 as GeometryType).coordinates;
```
Comment (lukas-h) [inline lib/src/booleans/boolean_contains.dart:30]: can these be final?

File: lib/src/booleans/boolean_helper.dart [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+bool isPolygonInMultiPolygon(Polygon polygon1, MultiPolygon poligon2) =>
+    _isPolygonInGeoJsonPolygon(polygon1, poligon2);
+
+bool _isPolygonInGeoJsonPolygon(
+  Polygon polygon1,
+  GeoJSONObject polygon2,
+) {
+  var boundingBoxOfPolygon1 = bbox(polygon1);
+  var boundingBoxOfPolygon2 = bbox(polygon2);
```
Comment (lukas-h) [inline lib/src/booleans/boolean_helper.dart:201]: final?

File: lib/src/booleans/boolean_helper.dart [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+bool isLineInPolygon(LineString line, Polygon polygon) =>
+    _isLineInGeoJsonPolygon(line, polygon);
+
+bool isLineInMultiPolygon(LineString line, MultiPolygon polygon) =>
+    _isLineInGeoJsonPolygon(line, polygon);
+
+bool _isLineInGeoJsonPolygon(LineString line, GeoJSONObject polygon) {
+  var boundingBoxOfPolygon = bbox(polygon);
+  var boundingBoxOfLine = bbox(line);
```
Comment (lukas-h) [inline lib/src/booleans/boolean_helper.dart:121]: final?

File: lib/src/booleans/boolean_within.dart [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+/// );
+/// booleanWithin(point, line);
+/// //=true
+/// ```
+bool booleanWithin(
+  GeoJSONObject feature1,
+  GeoJSONObject feature2,
+) {
+  var geom1 = getGeom(feature1);
+  var geom2 = getGeom(feature2);
```
Comment (lukas-h) [inline lib/src/booleans/boolean_within.dart:31]: final?

File: lib/src/booleans/discussion.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+// booleanWithin(LineString, Polygon): TurfJs seems to have an bug. When the last
+// point of the line is outside of the polygon, the function returns true.
+// turf-boolean-within -> isLineInPoly
+
+```
+
+
+
+https://geojson.io/#map=4.84/4.83/2.16
+
```
Comment (lukas-h) [inline lib/src/booleans/discussion.md:23]: should this be in the merged code or should this better be in an actual github issue / discussion?

File: test/booleans/within_test.dart [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+          [0, 0],
+          [0, 100],
+          [100, 100],
+          [100, 0],
+          [0, 0],
+        ],
+      ]);
+      var pointIn = point([50, 50]);
+      var pointOut = point([140, 150]);
+
```
Comment (lukas-h) [inline test/booleans/within_test.dart:73]: there are a lot of variables that at first glance could very well be `final`

File: test/booleans/within_test.dart [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+  for (var file in testDirectory.listSync(recursive: true)) {
+    if (file is File && file.path.endsWith('.geojson')) {
+      if (file.path.contains('skip')) continue;
+
+      final content = file.readAsStringSync();
+      final geoJson = GeoJSONObject.fromJson(jsonDecode(content));
+      test(file.path, geoJson);
+    }
+  }
+}
```
Comment (lukas-h) [inline test/booleans/within_test.dart:122]: Idea: let's create a helper file with all these functions, so they can be shared between tests

**Review by lukas-h — CHANGES_REQUESTED** [review]
basically just a couple of variables can be `final` – lgtm otherwise

File: lib/src/booleans/discussion.md (hunk 6 above)
Comment (jsiedentop) [reply lib/src/booleans/discussion.md:23]: It was actually not planned to push this file. I missed to delete this file.

File: test/booleans/within_test.dart (hunk 8 above)
Comment (jsiedentop) [reply test/booleans/within_test.dart:122]: Already happened in the context of lineOverlap / booleanOverlap.

**Comment by jsiedentop** [comment]
It's worth mentioning that the pipeline failed due to two analyzer errors. The errors are related to the recent Dart 3.3.0 release and also affect the main branch.

**Review by lukas-h — APPROVED** [review]
everything looks fine!
If CI/CD says the same, then let's go!!

**Pull request merged**